File: creational/singleton/Singleton.py

class Singleton:

	__instance = None

	def __new__(cls):
		if cls.__instance is None:
			cls.__instance = super().__new__(cls)
		return cls.__instance



singleton_obj1 = Singleton()
singleton_obj2 = Singleton()

print(singleton_obj1)
print(singleton_obj2)

File: creational/singleton/Singleton.js
var Singleton = (function () {
	var instance;

	function createInstance() {
		var obj = new Object();
		return obj;
	}

	return {
		getInstance: function() {
			if (!instance) {
				instance = createInstance();
			}
			return instance;
		}	
	};
})();

var one = Singleton.getInstance();
var two = Singleton.getInstance();
alert (one === two);
